api/authentication.py

```diff
@@ -0,0 +1,66 @@
+from datetime import datetime
+import json
+import requests
+
+from django.conf import settings
+from django.core.cache import cache
+
+from allauth.socialaccount.models import SocialAccount
+from rest_framework.authentication import (
+    BaseAuthentication, get_authorization_header
+)
+
+
+def get_cache_key(token):
+    return f'fxa_token_{token}'
+
+
+class FxaTokenAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        authorization = get_authorization_header(request).decode()
+        if not authorization or not authorization.startswith('Bearer '):
+            return None
+
+        token = request.headers.get('Authorization').split(' ')[1]
+        cache_key = get_cache_key(token)
+        fxa_resp = cache.get(cache_key)
+        if not fxa_resp:
```

Comment:
I think the cache hit cases are unchecked. I'd expect tests for:

* A 200 valid response with an active FxA user is cached
* A non-200 response is cached - I suspect this would break, since the status code is only checked on a request.